From: Eric Dumazet <edumazet@google.com>
Date: Thu, 19 Nov 2015 12:11:23 -0800
Subject: net: avoid NULL deref in napi_get_frags()
Patch-mainline: v4.5-rc1
Git-commit: e2f9dc3bd213792ac006e83f50a5453f23b8c354
References: bsc#966191 FATE#320230 bsc#966186 FATE#320228

napi_alloc_skb() can return NULL.
We should not crash should this happen.

Fixes: 93f93a440415 ("net: move skb_mark_napi_id() into core networking stack")
Signed-off-by: Eric Dumazet <edumazet@google.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 net/core/dev.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

--- a/net/core/dev.c
+++ b/net/core/dev.c
@@ -4398,8 +4398,10 @@ struct sk_buff *napi_get_frags(struct na
 
 	if (!skb) {
 		skb = napi_alloc_skb(napi, GRO_MAX_HEAD);
-		napi->skb = skb;
-		skb_mark_napi_id(skb, napi);
+		if (skb) {
+			napi->skb = skb;
+			skb_mark_napi_id(skb, napi);
+		}
 	}
 	return skb;
 }
